pkg/drivers/kic/kic.go

```diff
@@ -203,6 +208,15 @@ func (d *Driver) GetSSHPort() (int, error) {
 	return p, nil
 }
 
+// GetRegistryAddonPort returns port for use with registry
+func GetRegistryAddonPort(cc *config.ClusterConfig) (int, error) {
+	p, err := oci.ForwardedPort(cc.Driver, cc.Name, constants.RegistryAddonPort)
```

Comment:
this should use ForwardedPort
only if the goos. is NOT linux and also IF driver is Kic
use 
driver.IsKIC(cc.Driver) to check if driver is KIC

Reply:
no need to have this function here, just handle it in the same place u call it. bellow